src/providers/ConnectorState/LogLoop.ts

```diff
@@ -24,31 +28,34 @@ export class LogLoop {
     this.loopId = uniqueId()
   }
 
-  async sendLogs(client: CozyClient, limit?: number): Promise<void> {
+  async sendLogs(client: CozyClient): Promise<void> {
     try {
       const state = store.getState()
-      log.debug(
-        `🐙 ${this.loopId} - Current number of logs to handle: `,
-        state.connectorLogs.logs.length
-      )
-
-      const logs = limit
-        ? state.connectorLogs.logs.slice(0, limit)
-        : state.connectorLogs.logs
+      const logs = state.connectorLogs.logs
 
-      log.debug(`🐟 ${this.loopId} - logs`, logs)
+      // You can activate this line to get the log loop data
+      //log.debug(`🐟 ${this.loopId} - logs`, logs)
 
-      store.dispatch(limit ? spliceLogs(limit) : clearLogs())
-      const state2 = store.getState()
-
-      await sleep(500)
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-unsafe-member-access
-      log.debug(`🐙 ${this.loopId} - ${client.getStackClient().uri}`)
-
-      log.debug(
-        `🐙 ${this.loopId} - New number of logs to handle: `,
-        state2.connectorLogs.logs.length
-      )
+      const slugs = Object.keys(logs)
+      for (const slug of slugs) {
+        log.debug(`Sending ${slug} logs batch`)
+        // Clean slug property in LogObj
+        const cleanedLogs = logs[slug]?.map(({ slug, ...el }) => el)
+        if (!cleanedLogs) {
+          continue
+        }
+        try {
+          // Sending
+          await client
+            .getStackClient()
+            .fetchJSON('POST', `/konnectors/${slug}/logs`, cleanedLogs)
+          // Deleting
+          store.dispatch(removeLogs({ slug, number: cleanedLogs.length }))
+        } catch (e) {
+          log.error(`Error while sending ${slug} logs`)
+          log.error(e)
```

Comment:
Is it ok to log the error object directly? What will happen if the reason is itself an object?

Reply:
It seems better than nothing. What do you recommend instead?